enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/StoreCopyRequestHandler.java

```diff
@@ -111,6 +116,26 @@ else if ( !isTransactionWithinReach( request.requiredTransactionId(), checkpoint
         }
     }
 
+    private boolean isTransactionWithinReach( long transactionId )
+    {
+        return checkpointerSupplier.get().lastCheckPointedTransactionId() >= transactionId;
+    }
+
+    private void asyncCheckpoint() throws IOException
+    {
+        checkpointThreadPool.execute( () ->
```

Comment:
Would this not still queue up a bunch of runnables ? If a checkpoint i already running we should not need to do anything.

Reply:
Addressed by introducing a locking executor